clients/src/main/java/org/apache/kafka/common/requests/OffsetCommitResponse.java

```diff
@@ -50,14 +52,21 @@
 public class OffsetCommitResponse extends AbstractResponse {
 
     private final OffsetCommitResponseData data;
+    private final short version;
```

Comment:
Adding the version to the response seems to be an anti-pattern as I haven't seen any other similar use in other responses. Semantically it should be OK because the response instance is supposed to be built against a given version. If another approach is advisable, I will remove it.

Reply:
Did you check how we did this for the FetchRequest?